include/9on12Util.h

```diff
@@ -35,6 +35,61 @@ namespace D3D9on12
         }
     };
 
+    class Result
+    {
+    public:
+        enum Value : uint8_t
+        {
+            S_SUCCESS = 0x00,
+            S_CHANGE = 0x01,
+            E_FAILURE = 0x10,
+            E_INVALID_ARG = 0x20
+        };
+
+        Result() = default;
+        constexpr Result(Value result) : value(result) {}
+
+        //Prevent usage: if(result)
+        explicit operator bool() = delete;
+        explicit operator HRESULT() = delete;
+        constexpr Result operator||(const Result result) { return result.value > value ? result : *this; }
+
+        constexpr bool Succeeded() const { return value < Result::E_FAILURE; }
+        constexpr bool Failed() const { return value >= Result::E_FAILURE; }
+        constexpr bool operator==(Result other) const noexcept { return value == other.value; }
+        constexpr bool operator!=(Result other) const noexcept { return value != other.value; }
+
+        //Get the underlying enum value. This allows nice switch syntax without enabling implicit casting of Result to HRESULT
+        const Value AsEnum() { return value; }
+        //Explicit way of translating to HRESULT for communicating with parts of the codebase that haven't been migrated from HRESULT.
+        const HRESULT AsHresult() 

```

Comment:
For all the functions here: I think your `const` was meant to be after the argument list. And you can probably promote them all to `constexpr` (which does go first).